src/backend/distributed/shared_library_init.c

```diff
@@ -1100,6 +1104,23 @@ RegisterCitusConfigVariables(void)
 		GUC_NO_SHOW_ALL,
 		NULL, NULL, NULL);
 
+	DefineCustomStringVariable(
+		"citus.hide_shards_from_app_name_prefixes",
+		gettext_noop("If application_name starts with one of these values, hide shards"),
+		gettext_noop("Citus places distributed tables and shards in the same schema. "
+					 "That can cause confusion when inspecting the list of tables on "
+					 "a node with shards. This GUC can be used to hide the shards from "
+					 "pg_class for certain applications based on the application_name "
+					 "of the connection. This behaviour can be overridden using the "
+					 "citus.override_table_visibility setting"),
+		&HideShardsFromAppNamePrefixes,
+		"psql,pgAdmin,pg_dump",
+		PGC_SUSET,
```

Comment:
For the defaults, I have few  questions/notes:

1) Unfortunately, Azure Data Studio by default doesn't set the app name, it is empty string. But, the users can set it via the portal. I wonder if empty string app names are common. If so, we could maybe add that as well somehow?

2) One question raised during MX sync was, what if we set it `*` by default? I guess our answer would be we might break existing apps dealing with the shards on the workers?

3) I guess `citus.override_table_visibility` becomes not useful anymore? As we approach a major release, maybe we should drop it?

4) Seems useful if non-super users can change this for local session? In general, if a user wants to change this on HSC, we should allow them via portal?

Reply:
1. I could perhaps add an additional GUC that is not prefixed
2. I would be weary of breaking all sorts of monitoring software
3. sure
4. makes sense